src/pathology-report/ReportComponent.tsx

```diff
@@ -1,40 +1,476 @@
-import React from 'react';
-import styles from './ReportComponent.css';
-import { TableRow } from './TableRow';
-import { getPatientEncounters } from './ReportComponent.resource';
+import React from "react";
+import cloneDeep from "lodash-es/cloneDeep";
+import {
+  DataTable,
+  Table,
+  TableHead,
+  TableRow,
+  TableHeader,
+  TableBody,
+  TableCell,
+} from "carbon-components-react";
+
+import styles from "./ReportComponent.css";
+// import { TableRow } from './TableRow';
+import {
+  getUserLocation,
+  getConceptAnswers,
+  getLocations,
+  getEncounters,
+  postSampleDropoffObs,
+  voidSampleDropoff,
+  postSampleStatusChangeObs,
+  updateSampleStatusChangeObs,
+  postReferralStatusChangeObs,
+  updateReferralStatusChangeObs,
+} from "./ReportComponent.resource";
+import {
+  RWINKWAVUHEALTHCENTERNAME,
+  SAMPLESTATUSUUID,
+  REFERRALSTATUSUUID,
+  SAMPLEDROPOFFUUID,
+  YESCONCEPTNAME,
+} from "../constants.js";
 
 const ReportComponent = () => {
+  const [userLocation, setUserLocation] = React.useState();
   const [encountersList, setEncountersList] = React.useState([]);
-  // const [selectedEncounter, setSelectedEncounter] = React.useState();
+  const tempEncList = React.useRef([]);
```

Comment:
This is an unusual use of `useRef` and I really don't think you need it at all. It looks like at the top of each function that uses `tempEncList` you are re-initializing it to something new. That indicates to me that it can just live as a `const` within each of those functions—it doesn't need to exist outside those functions at all.

Reply:
```suggestion
```